sdl_android/src/main/java/com/smartdevicelink/util/AndroidTools.java

```diff
@@ -106,9 +106,10 @@ public static boolean isServiceExported(Context context, ComponentName name) {
 					try {
 						packageInfo = packageManager.getPackageInfo(info.serviceInfo.packageName, 0);
 					} catch (NameNotFoundException e) {
-					} finally {
-						sdlAppInfoList.add(new SdlAppInfo(info, packageInfo));
+						//Package was not found, likely a sign the resolve info can't be trusted.
 					}
+					sdlAppInfoList.add(new SdlAppInfo(info, packageInfo));
```

Comment:
This change will still add the bad package. It should be:
```
try {
 						packageInfo = packageManager.getPackageInfo(info.serviceInfo.packageName, 0);	 						packageInfo = packageManager.getPackageInfo(info.serviceInfo.packageName, 0);
sdlAppInfoList.add(new SdlAppInfo(info, packageInfo));
 					} catch (NameNotFoundException e) {	 					} catch (NameNotFoundException e) { // comments here
-					} 
+					
```

Reply:
Code under a `catch` block is not executed, that is why there is the `finally` block.

Reply:
There are more than one way to do this. I know my change works. The code that adds the package cannot be outside after the catch block this way. It will be executed when the catch block is done. In this case, it's almost similar to having it in a finally clause.